internal/k8s/status.go

```diff
@@ -638,12 +639,11 @@ func (su *statusUpdater) UpdatePolicyStatus(pol *v1.Policy, state string, reason
 		return nil
 	}
 
-	polCopy := polLatest.(*v1.Policy)
-
-	if !hasPolicyStatusChanged(polCopy, state, reason, message) {
+	if !hasPolicyStatusChanged(polLatest, state, reason, message) {
 		return nil
 	}
 
+	polCopy := polLatest.DeepCopy()
```

Comment:
this looks like a bug fix. would it make sense to put it separately?

Reply:
I can fix it now as part of this, or I can log a bug, put it on the backlog, discuss it in sprint planning + backlog review, create a PR, wait a few days and merge it.

Reply:
Well, I suggest looking at it from the reviewer perspective. As a submitter you create a refactor PR, along the way there is an unrelated code change. For the reviewer, it appears as a bug and  brings different questions: what is that change about? why is it part of the refactor? is it a bug? Is anyone affected? Do we need to mention it in the release notes. That change came out of the blue.

in any case, when making a change like that, better to do it separately - a PR or at least a commit. and include its own rationale. Such changes can also introduce bugs, so this way it will be easier to track it down.

Reply:
grand so, removed.